src/proto/streams/flow_control.rs

```diff
@@ -136,16 +136,16 @@ impl FlowControl {
     ///
     /// This is called after receiving a SETTINGS frame with a lower
     /// INITIAL_WINDOW_SIZE value.
-    pub fn dec_send_window(&mut self, sz: WindowSize) -> Result<(), Reason> {
+    pub fn dec_send_window(&mut self, sz: WindowSize) {
         tracing::trace!(
             "dec_window; sz={}; window={}, available={}",
             sz,
             self.window_size,
             self.available
         );
-        // ~~This should not be able to overflow `window_size` from the bottom.~~ wrong. it can.
-        self.window_size.decrease_by(sz)?;
-        Ok(())
+        self.window_size
+            .decrease_by(sz)
+            .expect("window_size should be greater than decrement");
```

Comment:
Why the change to make this method panic instead of returning an error?

Reply:
Well it's not supposed to happen ever, and it used to be an assertion inside. At the very least it shouldn't be `FLOW_CONTROL_ERROR`.

Reply:
Ideally most of those `?` in flow control internals would be `INTERNAL_ERROR` with debug payloads explaining that a bug happened, imo.

Reply:
Would you prefer I put the `expect` at the call site, and not change the signature of `dec_send_window`?

Reply:
Well, looking at the changes here, it still seems like it's possible. It might not be _right_ to happen, but if it does, shouldn't a connection close be better than a panic?

Or is the change in `send.rs` truly the last place that could ever make this be wrong?

Reply:
I made it return an `INTERNAL_ERROR`, we can revisit later all of that to panic in debug builds.

Reply:
Turns out I forgot to push the part where I made it `INTERNAL_ERROR`, oh well.